gemini/use-cases/retrieval-augmented-generation/multimodal_rag_langchain.ipynb

```diff
@@ -0,0 +1,1026 @@
+    "            base64_image = encode_image(img_path)\n",
+    "            img_base64_list.append(base64_image)\n",
+    "            image_summaries.append(image_summarize(base64_image, prompt))\n",
+    "\n",
+    "    return img_base64_list, image_summaries\n",
+    "\n",
+    "\n",
+    "# Image summaries\n",
+    "img_base64_list, image_summaries = generate_img_summaries(\".\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create & Deploy Vertex AI Vector Search Index & Endpoint\n",
+    "\n",
+    "Skip this step if you already have Vector Search set up.\n",
+    "\n",
+    "- https://console.cloud.google.com/vertex-ai/matching-engine/indexes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- Create [`MatchingEngineIndex`](https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform.MatchingEngineIndex)\n",
+    "  - https://cloud.google.com/vertex-ai/docs/vector-search/create-manage-index"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/text-embeddings\n",
+    "DIMENSIONS = 768  # Dimensions output from textembedding-gecko\n",
+    "\n",
+    "index = aiplatform.MatchingEngineIndex.create_tree_ah_index(\n",
+    "    display_name=\"mm_rag_langchain_index\",\n",
+    "    dimensions=DIMENSIONS,\n",
+    "    approximate_neighbors_count=150,\n",
+    "    leaf_node_embedding_count=500,\n",
+    "    leaf_nodes_to_search_percent=7,\n",
+    "    description=\"Multimodal RAG LangChain Index\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- Create [`MatchingEngineIndexEndpoint`](https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform.MatchingEngineIndexEndpoint)\n",
+    "  - https://cloud.google.com/vertex-ai/docs/vector-search/deploy-index-public"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DEPLOYED_INDEX_ID = \"mm_rag_langchain_index_endpoint\"\n",
+    "\n",
+    "index_endpoint = aiplatform.MatchingEngineIndexEndpoint.create(\n",
+    "    display_name=DEPLOYED_INDEX_ID,\n",
+    "    description=\"Multimodal RAG LangChain Index Endpoint\",\n",
+    "    public_endpoint_enabled=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- Deploy Index to Index Endpoint\n",
+    "  - NOTE: You can stop this cell after starting it instead of waiting for deployment\n",
+    "  - You can check the status at https://console.cloud.google.com/vertex-ai/matching-engine/indexes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "index_endpoint = index_endpoint.deploy_index(\n",
```

Comment:
Should probably add a note that this deploy index command will take a while to run, on the order of 10-15 mins when I ran just now.

Reply:
Added